src/test-run/index.js

```diff
@@ -91,7 +93,9 @@ export default class TestRun extends Session {
         this.resolveWaitForFileDownloadingPromise = null;
 
         this.debugging               = this.opts.debugMode;
+        this.debugOnFail             = this.opts.debugOnFail;
         this.disableDebugBreakpoints = false;
+        this.debugReporterPluginHost = new ReporterPluginHost(testcafeReporterSpecPlugin);
```

Comment:
Requiring `spec` reporter only for using `formatError` helper seems like a bit overkill for me. Personally I would prefer to use `new ReporterPluginHost(null)` or `new ReporterPluginHost()`, maybe with comment that we aren't going to actually use reporter plugin, and only want to use formatting helpers. Or maybe it's even better to extract formatting helpers to a separate class like a `Formatter`.

Reply:
nice point